src/nested-row-expanders.js

```diff
@@ -0,0 +1,102 @@
+import './nested-row-expanders.css'
+import { functor } from '@zambezi/fun'
+import { select, event } from 'd3-selection'
+import { selectionChanged } from '@zambezi/d3-utils'
+import { unwrap } from './unwrap-row'
+
+const collapse = 'is-collapse'
+    , expand = 'is-expand'
+    , upRight = 'is-up-right'
+    , verticalRight = 'is-vertical-right'
+    , horizontal = 'is-horizontal'
+    , vertical = 'is-vertical'
+    , classes = [
+        collapse, expand, upRight, verticalRight, horizontal, vertical
+      ]
+    , changed = selectionChanged()
+
+export function createNestedRowExpanders() {
+
+  function nestedRowExpanders(d, i) {
+
+    const cell = d3.select(this).classed('nested-expander-cell', true)
+        , column = d.column
+        , value = d.value
+        , row = d.row
+        , parentLinesData = buildParentLineData(row).concat(column.format(value))
+        , update = cell.select(changed.key(functor(parentLinesData.join('×'))))
+            .selectAll('.nested-indicator, .nested-text-field, .formatted-text')
+            .data(parentLinesData)
+        , enter = update.enter().append('span')
+        , exit = update.exit().remove()
+        , merge = update.merge(enter)
+
+    merge.classed('formatted-text', false)
+        .each(updateNestedIndicatorClasses)
+      .filter(isNestedIndicator)
```

Comment:
This is common D3 practice, on chaining methods, indent differently when the operator returns a new selection so it can be know at a glance on which selection the stuff is being run.
It's spread on several "best practices" snippets I've seen here and there, but can be heard from the main man,
 
> The recommended indentation pattern for method chaining is four spaces for methods that preserve the current selection and two spaces for methods that change the selection.
> --- https://bost.ocks.org/mike/bar/#chaining

A now classic example, with filtering in particular, can be found here,
https://gist.github.com/d3noob/8dc93bce7e7200ab487d#file-index-html-L82